### PR TITLE
CBG-2880: Cancel timed out gocbcore dcp operations

### PR DESCRIPTION
**Pull request by chvck**
When timeouts occur the gocbcore operations must be cancelled to flush them from the agent's system. Not doing so can result in handlers being called unexpectedly, e.g. ending up writing to a channel that is no longer being read from.

CBG-2880

Describe your PR here...
- Use bullet points if there's more than one thing changed

## Pre-review checklist
- [ x ] Removed debug logging (`fmt.Print`, `log.Print`, ...)
- [ x ] Logging sensitive data? Make sure it's tagged (e.g. `base.UD(docID)`, `base.MD(dbName)`)
- [ x ] Updated relevant information in the API specifications (such as endpoint descriptions, schemas, ...) in `docs/api`

## Dependencies (if applicable)
- [ x ] Link upstream PRs
- [ x ] Update Go module dependencies when merged

## [Integration Tests](https://jenkins.sgwdev.com/job/SyncGateway-Integration/build?delay=0sec)
- [ ] `GSI=true,xattrs=true` https://jenkins.sgwdev.com/job/SyncGateway-Integration/000/

### CODE REVIEW

**Comment by chvck** [comment]
I've introduced another bug here. Will push a fix shortly.

File: base/dcp_client.go [hunk 1]
```diff
@@ -195,6 +197,8 @@ func (dc *DCPClient) getCollectionHighSeqNos(collectionID uint32) ([]uint64, err
 				return nil, err
 			}
 		case <-time.After(getVbSeqnoTimeout):
+			op.Cancel()
+			<-highSeqNoError
```
Comment (adamcfraser) [inline base/dcp_client.go:201]: @chvck Is this still required now that this is a buffered channel?  (same below for openStreamError)

Comment (chvck) [reply base/dcp_client.go:201]: Possibly not, no. Just ensures the value is read and everything is tidied up before returning. Could also read the error value from it, and return success if there is no error - possible if Cancel and successful response return are racing.

Comment (adamcfraser) [reply base/dcp_client.go:201]: With the additional information that op.Cancel() is always going to invoke the callback, the current approach makes sense.

**Review by adamcfraser — APPROVED** [review]

**Pull request merged**